uniflow-core/src/main/kotlin/io/uniflow/core/flow/data/UIState.kt

```diff
@@ -24,5 +24,26 @@ open class UIState : UIData {
     object Empty : UIState()
     object Loading : UIState()
     object Success : UIState()
-    data class Failed(val message: String? = null, val error: Throwable? = null, val state: UIState? = null) : UIState()
+    data class Failed(val message: String? = null, val error: Throwable? = null, val state: UIState? = null) : UIState() {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
```

Comment:
Can't this be simplified to `if (other !is Failed)`?

Reply:
yes, better